job-server/src/spark.jobserver/JobServer.scala

```diff
@@ -55,12 +55,21 @@ object JobServer {
     val clazz = Class.forName(config.getString("spark.jobserver.jobdao"))
     val ctor = clazz.getDeclaredConstructor(Class.forName("com.typesafe.config.Config"))
     try {
+      val contextPerJvm = config.getBoolean("spark.jobserver.context-per-jvm")
+      // Check if we are using correct DB backend when context-per-jvm is enabled.
+      // JobFileDAO and H2 mem is not supported.
+      if (contextPerJvm) {
+        if (clazz.getName != "spark.jobserver.io.JobSqlDAO") {
```

Comment:
I think instead of whitelisting JobSqlDao, we could blacklist the JobFileDao.   There is a PR for a Cassandra DAO, which would not be a SqlDao I think.